### PR TITLE
Cleaner and consistent design of front-end footer

### PR DESCRIPTION
**Pull request by drummer83**
#### What? Why?

In the current footer design I found four things which are not perfectly designed:

1. Tiny font for important links below 'Keep in touch', 'Navigate' and 'Join us'
2. Misaligned rows between 'Keep in touch' and 'Navigate'
3. Different font sizes of text and link below 'Join us'
4. Different font size of last paragraph in the legal section

footer old

#### What should we test?

- Re 1: Font size of links is slightly increased
- Re 2: Rows are well aligned on all relevant screen sizes (unless there is a line break)
- Re 2: Rows are well aligned also if no social icons are shown (to be configured in admin/config/content)
- Re 3: Text and link below 'Join us' have the same font size
- Re 4: Text of last paragraph in the legal section has the same font size as the other legal paragraphs
- General: Responsive behaviour is still working as expected
- General: Design is good on all relevant screen sizes

This is what it should look like:

Footer

#### Release notes
Cleaner and consistent design of front-end footer

Changelog Category: User facing changes

### CODE REVIEW

File: app/assets/stylesheets/darkswarm/footer.scss [hunk 1]
```diff
@@ -5,7 +5,7 @@
 footer {
   .row {
     p a {
-      font-size: 0.875rem;
+      font-size: 1rem;
```
Comment (sauloperez) [inline app/assets/stylesheets/darkswarm/footer.scss:8]: I wonder if we really need to set the font-size of links within paragraphs. They should inherit the same size as the paragraph.

File: app/views/shared/_footer.html.haml [hunk 2]
```diff
@@ -31,25 +31,26 @@
...
-          - if ContentConfig.footer_linkedin_url.present?
-            %a{href: ContentConfig.footer_linkedin_url}
-              %i.ofn-i_042-linkedin
-          - if ContentConfig.footer_googleplus_url.present?
-            %a{href: ContentConfig.footer_googleplus_url}
-              %i.ofn-i_046-g
-          - if ContentConfig.footer_pinterest_url.present?
-            %a{href: ContentConfig.footer_pinterest_url}
-              %i.ofn-i_045-pintrest
+        - if ContentConfig.footer_facebook_url.present? || ContentConfig.footer_twitter_url.present? || ContentConfig.footer_instagram_url.present? || ContentConfig.footer_linkedin_url.present? || ContentConfig.footer_googleplus_url.present? || ContentConfig.footer_pinterest_url.present?
```
Comment (sauloperez) [inline app/views/shared/_footer.html.haml:34]: we should rather extract a helper method from this long line and put it in `app/helpers/footer_links_helper.rb`.

**Review by sauloperez — CHANGES_REQUESTED** [review]

**Comment by sauloperez** [comment]
It looks like this broke a test. We should make it a bit more robust, not relying on the markup so much.

File: app/assets/stylesheets/darkswarm/footer.scss (hunk 1 above)
Comment (drummer83) [reply app/assets/stylesheets/darkswarm/footer.scss:8]: Thanks @sauloperez! I was wondering too - but it was there before and it works. I am happy to remove it. But then we need an #id for the "Secure and trusted" headline to still have it bigger (1.5 rem) than the text (1 rem).
Which solution do you prefer?

**Comment by drummer83** [comment]
> It looks like this broke a test. We should make it a bit more robust, not relying on the markup so much.

@sauloperez: The failing test seems to be about the cookie banner. I don't think this has something to do with my changes. There is even an issue number assigned: #2599 (closed)
`rspec ./engines/web/spec/features/consumer/cookies_spec.rb:37 # Cookies banner in the homepage does not show after cookies are accepted, and policy page is opened through the footer, and closed again (bug #2599)
`

File: app/views/shared/_footer.html.haml [hunk 3]
```diff
@@ -107,10 +108,9 @@
         %p.text-small
           = t '.footer_legal_text_html', {content_license: link_to('CC BY-SA 3.0', 'https://creativecommons.org/licenses/by-sa/3.0/'), code_license: link_to('AGPL 3', 'https://tldrlegal.com/license/gnu-affero-general-public-license-v3-(agpl-3.0)' )}
         %p.text-small
-          %div
```
Comment (sauloperez) [inline app/views/shared/_footer.html.haml:110]: If you look at the failed test's output @drummer83 

> Failure/Error: find("div > a", text: "cookies policy").click
     
     Capybara::ElementNotFound:
       Unable to find css "div > a"
     # ./engines/web/spec/features/consumer/cookies_spec.rb:158:in `click_footer_cookies_policy_link_and_wait'
     # ./engines/web/spec/features/consumer/cookies_spec.rb:41:in `block (4 levels) in <top (required)>'
     # ./engines/web/spec/features/consumer/cookies_spec.rb:10:in `block (3 levels) in <top (required)>'

This is the root cause. We need to decouple the test from the markup so it doesn't break.

Comment (drummer83) [reply app/views/shared/_footer.html.haml:110]: Thank you, @sauloperez!
Sorry, I am newbie...

**Review by sauloperez — APPROVED** [review]

**Review by jibees — APPROVED** [review]
Thanks!

**Comment by filipefurtad0** [comment]
Hey @drummer83 ,

Thanks for these improvements! The screenshots below aim to highlight the changes this PR introduces: Before this PR (above), and after this PR (below).

image

image


Thank you for the detailed notes on what to check :+1: 

   - Re 1: Font size of links is slightly increased - OK
   - Re 2: Rows are well aligned on all relevant screen sizes (unless there is a line break) - OK; of course this alignment might be lost if other languages and narrower screen sizes are used, ex. below (iPhone, Spanish), but I think this is fine?

image


   - Re 2: Rows are well aligned also if no social icons are shown (to be configured in admin/config/content) - OK, looks good as well without social media icons:
 
image

   - Re 3: Text and link below 'Join us' have the same font size - OK
   - Re 4: Text of last paragraph in the legal section has the same font size as the other legal paragraphs - OK
   - General: Responsive behaviour is still working as expected - changed screen size in desktop browsers Firefox/Chrome - OK
   - General: Design is good on all relevant screen sizes -> checked in latest iPhone and Android mobile devices (Browserstack) - OK; below (Android 12, French),

image

Again, there is some misalignment introduced, but this was the case before as well, and you PR brings plenty of improvements, so - moving to ready to go! Thanks for another PR Konrad :muscle:

**Comment by sauloperez** [comment]
Das is der Hammer @drummer83 :tada: !

**Pull request merged**